django_mongodb/aggregates.py

```diff
@@ -0,0 +1,75 @@
+from copy import deepcopy
+
+from django.db.models.aggregates import Aggregate, Count, StdDev, Variance
+from django.db.models.expressions import Case, Value, When
+from django.db.models.lookups import Exact
+from django.db.models.sql.where import WhereNode
+
+from .query_utils import process_lhs
+
+MONGO_AGGREGATIONS = {Count: "sum"}
+
+
+def aggregate(self, compiler, connection, operator=None, **extra_context):  # noqa: ARG001
+    if self.filter:
+        node = self.copy()
+        node.filter = None
+        source_expressions = node.get_source_expressions()
+        condition = When(self.filter, then=source_expressions[0])
+        node.set_source_expressions([Case(condition)] + source_expressions[1:])
+    else:
+        node = self
+    lhs_mql = process_lhs(node, compiler, connection)
+    operator = operator or MONGO_AGGREGATIONS.get(self.__class__, self.function.lower())
+    return {f"${operator}": lhs_mql}
+
+
+def count(self, compiler, connection, resolve_inner_expression=False, **extra_context):  # noqa: ARG001
+    """
+    When resolve_inner_expression is True, return the argument as MQL that
+    resolves as a value. This is used to count different elements, so the inner
+    values are returned to be pushed into a set.
+    """
+    if not self.distinct or resolve_inner_expression:
+        if self.filter:
+            node = self.copy()
+            node.filter = None
+            source_expressions = node.get_source_expressions()
+            filter_ = deepcopy(self.filter)
+            filter_.add(
+                WhereNode([Exact(source_expressions[0], Value(None))], negated=True),
```

Comment:
Sorry @timgraham I deleted your comment by mistake, the answer is:
The count only counts values if they aren't none. if the expression result is a string, number or something it sums as 1. Looking again the code, maybe there is a bug when there is filter and distinct options. Will check.

Reply:
Ok, there are like 3 places with nulls. 😬 or maybe two.
The example when this code is use is:
```
Select count(name) filter (where surname = 'Lupi')
from T1
```
Here we have to sum of the elements that fulfill the filter, we handle it with a case and the first (and only) source_expression would be the transformed (idk if there is others, but I just copy the mechanism from source)

I will explain it one by one.

But we have to change the exact(value, None) for IsNull(value, True) they are not the same.